src/Compilers/CSharp/Portable/Binder/Binder_Symbols.cs

```diff
@@ -294,37 +294,38 @@ internal NamespaceOrTypeOrAliasSymbolWithAnnotations BindNamespaceOrTypeOrAliasS
                     {
                         TypeSyntax typeArgumentSyntax = ((NullableTypeSyntax)syntax).ElementType;
                         TypeSymbolWithAnnotations typeArgument = BindType(typeArgumentSyntax, diagnostics, basesBeingResolved);
-
-                        // TODO: NullableReferenceTypes - If nullable reference types aren't enabled, we can create Nullable<typeArgument> right here as an optimization. 
-                        //                                It is also safe to make a decision whether this is a nullable reference type or a Nullable<T> right here
-                        //                                if ShouldCheckConstraints is true (we are checking constructedType.IsNullableType() right here anyway).   
-                        TypeSymbolWithAnnotations constructedType = typeArgument.AsNullableReferenceOrValueType(Compilation, syntax.GetReference());
-                        if (ShouldCheckConstraints)
+                        TypeSymbolWithAnnotations constructedType;
+                        if (Compilation.IsFeatureEnabled(MessageID.IDS_FeatureStaticNullChecking))
```

Comment:
It feels strange that the feature flag would affect binding.

Reply:
Maybe add a comment that this is an optimization?

---
In reply to: [171062278](https://github.com/dotnet/roslyn/pull/25087#discussion_r171062278) [](ancestors = 171062278)